crypto/x509/x509_vfy.c

```diff
@@ -362,6 +362,20 @@ int X509_verify_cert(X509_STORE_CTX *ctx) {
         ok = 0;
         goto end;
       }
+
+      // OpenSSL 1.1.1 continuously re-checks for trust and breaks the loop
+      // as soon as trust has been established. 1.0.2 builds the chain with all
+      // possible certs first and only checks for trust if the final cert in
+      // the chain is self-signed.
+      // This caused additional unanticipated certs to be in the established
+      // certificate chain, particularly when |X509_V_FLAG_PARTIAL_CHAIN| was
+      // set. We try checking continuously for trust here for better 1.1.1
+      // compatibility.
+      trust = check_trust(ctx);
```

Comment:
line 383 is now double-work(?).

But you still need it for the SS (true CA) case. Dunno whether intermediate CA trust is the majority case or not for our applications.

Reply:
Yeah that was one of the reasons I left it there. Even if having a SS root wasn't the majority case, the implicit default of this function was to have one and `check_trust` on L383 would help check the default scenario.

The other reason was because X509 chain building/verification has too many subtle nuances for me to feel comfortable moving too many pieces around.. We could rearrange the for loop to not do duplicate work for `X509_V_FLAG_PARTIAL_CHAIN`, but using an early call to break into our existing logic seemed like the safest and most straightforward path.